File: Form43.vb
Imports System.ComponentModel

Public Class Form43
    Dim form As Form33_ColorBasedDropDownList = Nothing

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        form = New Form33_ColorBasedDropDownList
        form.Show()
        Me.Dispose()
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click

        form_flag = False
        Me.Dispose()
    End Sub

    Private Sub CheckBox1_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox1.CheckedChanged
        If CheckBox1.Checked = True Then
            sessionflag1 = False
        Else
            sessionflag1 = True
        End If
    End Sub

    Private Sub Form43_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub Form43_Closing(sender As Object, e As CancelEventArgs) Handles Me.Closing
        form_flag = False
    End Sub

    Private Sub Form43_Disposed(sender As Object, e As EventArgs) Handles Me.Disposed
        form_flag = False
    End Sub
End Class

File: Form43.cs
using System;
using System.ComponentModel;

namespace VSTO_Addins
{

    public partial class Form43
    {
        private Form33_ColorBasedDropDownList form = null;

        public Form43()
        {
            InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            form = new Form33_ColorBasedDropDownList();
            form.Show();
            Dispose();
        }

        private void Button2_Click(object sender, EventArgs e)
        {

            GlobalModule.form_flag = false;
            Dispose();
        }

        private void CheckBox1_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox1.Checked == true)
            {
                GlobalModule.sessionflag1 = false;
            }
            else
            {
                GlobalModule.sessionflag1 = true;
            }
        }

        private void Form43_Load(object sender, EventArgs e)
        {

        }

        private void Form43_Closing(object sender, CancelEventArgs e)
        {
            GlobalModule.form_flag = false;
        }

        private void Form43_Disposed(object sender, EventArgs e)
        {
            GlobalModule.form_flag = false;
        }
    }
}